contracts/ERC20ExtendedToken.sol

```diff
@@ -34,21 +34,30 @@ contract ERC20ExtendedToken is DSTokenBase(0), DSAuth, ERC20Extended {
     decimals = _decimals;
   }
 
-  function mint(uint wad) public
-  auth
-  {
-    _balances[msg.sender] = add(_balances[msg.sender], wad);
-    _supply = add(_supply, wad);
-
-    emit Mint(msg.sender, wad);
-    emit Transfer(address(0x0), msg.sender, wad);
+  function mint(uint wad) public auth {
+    mint(msg.sender, wad);
   }
 
   function burn(uint wad) public {
-    _balances[msg.sender] = sub(_balances[msg.sender], wad);
-    _supply = sub(_supply, wad);
+    burn(msg.sender, wad);
+  }
 
-    emit Burn(msg.sender, wad);
+  function mint(address guy, uint wad) public auth {
+    _balances[guy] = add(_balances[guy], wad);
+    _supply = add(_supply, wad);
+    emit Mint(guy, wad);
+    emit Transfer(address(0x0), guy, wad);
   }
-}
 
+  function burn(address guy, uint wad) public {
+    if (guy != msg.sender && _approvals[guy][msg.sender] != uint(-1)) {
```

Comment:
What is the purpose of the second condition here? `_approvals[guy][msg.sender] != uint(-1)`?

Reply:
`uint(-1)` is equivalent to `UINT256_MAX` unless I'm mistaken, so we don't subtract the approval if they are maximum?

Reply:
This is related to the so-called binary approval in dappsys token implementation where you approve an address to manage _all_ your tokens via approving them to the max limit, see https://github.com/dapphub/ds-token/blob/master/src/token.sol#L37
So this line is saying that if you have such approval, you cannot use the `burn` function.

Reply:
Mm. That's definitely an undocumented dependency between these contracts then... can we introduce something like:

```
uint256 constant BINARY_APPROVAL = uint256(-1);
```

Which makes the semantics more clear?

Reply:
This is my last outstanding comment ^^

Reply:
I've completely removed our custom ERC20 implementation in the face of `ERC20ExtendedToken` and now using vanilla `DSToken` instead. Colony creators should be able to make their own informed choice as to which token contract to use, there's plenty on the market including `DSToken` which the DAI is based on and OpenZeppelin's `ERC20Mintable` token.